src/gui/qgstemporalcontrollerdockwidget.cpp

```diff
@@ -0,0 +1,238 @@
+
+  mStartDateTime->setDisplayFormat( locale.dateTimeFormat() );
+  mEndDateTime->setDisplayFormat( locale.dateTimeFormat() );
+
+  if ( range.begin().isValid() && range.end().isValid() )
+  {
+    mStartDateTime->setDateTime( range.begin() );
+    mEndDateTime->setDateTime( range.end() );
+  }
+
+  mSetToProjectTimeButton->setToolTip( tr( "Set to project time" ) );
+
+  QStringList listSteps = ( QStringList() << tr( "Seconds" ) << tr( "Minutes" ) << tr( "Hours" )
+                            << tr( "Days" ) <<
+                            tr( "Months" ) << ( "Years" ) );
+
+  mTimeStepsComboBox->addItems( listSteps );
+  mTimeStepsComboBox->setCurrentIndex( 2 );
+
+  mSpinBox->setMinimum( 1 );
+  mSpinBox->setSingleStep( 1 );
+  mSpinBox->setValue( 1 );
+  mSpinBox->setEnabled( true );
+
+  updateFrameRate();
+  updateTemporalExtent();
+  updateFrameDuration();
+}
+
+void QgsTemporalControllerDockWidget::updateFrameRate()
+{
+  if ( mSettingsDialog && mSettingsDialog->mapSettingsWidget() )
+    mNavigationObject->setFramesPerSeconds(
+      mSettingsDialog->mapSettingsWidget()->frameRateValue() );
+}
+
+void QgsTemporalControllerDockWidget::updateTemporalExtent()
+{
+  QgsDateTimeRange temporaExtent = QgsDateTimeRange( mStartDateTime->dateTime(),
+                                   mEndDateTime->dateTime() );
+  mNavigationObject->setTemporalExtents( temporaExtent );
+  mSlider->setRange( 0, mNavigationObject->totalFrameCount() - 1 );
+}
+
+void QgsTemporalControllerDockWidget::updateFrameDuration()
+{
+  mNavigationObject->setFrameDuration( interval( mTimeStepsComboBox->currentText(),
+                                       mSpinBox->value() ) );
+  mSlider->setRange( 0, mNavigationObject->totalFrameCount() - 1 );
+}
+
+void QgsTemporalControllerDockWidget::updateSlider( const QgsDateTimeRange &range )
+{
+  whileBlocking( mSlider )->setValue( mNavigationObject->currentFrameNumber() );
+  updateRangeLabel( range );
+}
+
+void QgsTemporalControllerDockWidget::updateRangeLabel( const QgsDateTimeRange &range )
+{
+  QLocale locale;
+  mCurrentRangeLabel->setText( tr( "Current temporal range is from %1 to %2" ).arg(
+                                 range.begin().toString( locale.dateTimeFormat() ),
+                                 range.end().toString( locale.dateTimeFormat() ) ) );
+}
+
+QgsTemporalNavigationObject *QgsTemporalControllerDockWidget::temporalController()
+{
+  return mNavigationObject;
+}
+
+void QgsTemporalControllerDockWidget::settingsDialog()
+{
+  mSettingsDialog = new QgsTemporalMapSettingsDialog( this );
+}
+
+void QgsTemporalControllerDockWidget::settings_clicked()
+{
+  mSettingsDialog->setVisible( !mSettingsDialog->isVisible() );
+}
+
+void QgsTemporalControllerDockWidget::timeSlider_valueChanged( int value )
+{
+  mNavigationObject->setCurrentFrameNumber( value );
+}
+
+QgsInterval QgsTemporalControllerDockWidget::interval( QString time, int value )
+{
+  QgsInterval interval;
+
+  if ( time == QString( "Seconds" ) )
```

Comment:
This should use the enum value mentioned above (in a switch statement)

(The current logic will break as soon as translations flow in)